packages/@lwc/engine/src/framework/__tests__/events.spec.ts

```diff
@@ -951,7 +951,7 @@ describe('Shadow Root events', () => {
             connectedCallback() {
                 this.template.addEventListener('click', evt => {
                     expect(evt.target.tagName).toBe('X-GRAND-CHILD');
-                    expect(evt.currentTarget.tagName).toBe('X-CHILD');
+                    expect(evt.currentTarget).toBe(this.template);
```

Comment:
this was also a bug in synthetic, the current target of an event listener added on the shadow must be the shadow itself. This is why we need to move these tests to karma, so we can check the behavior of the synthetic shadow.